[Strapi 5] Breaking change about removed support for some env-only options (#2007)
strapi/documentation: docusaurus/docs/dev-docs/migration/v4-to-v5/breaking-changes/removed-support-for-some-env-options.md

---
title: Some env-only configuration options are handled by the server configuration
description: In Strapi 5, some env-only configuration options are handled by the server configuration
displayed_sidebar: devDocsMigrationV5Sidebar
sidebar_label: Env options moved to server configuration
tags:
 - breaking changes
 - configuration
 - environment variables
 - server
---

import Intro from '/docs/snippets/breaking-change-page-intro.md'

# Some `env`-only configuration options are handled by the server configuration

> In Strapi 5, some configuration options that were only handled by environment variables are now handled in the [server configuration](/dev-docs/configurations/server) file.

Author: ```suggestion
In Strapi 5, some configuration options that were only handled by environment variables in Strapi v4 are now handled in the [server configuration](/dev-docs/configurations/server) file.
```

<Intro />

## Breaking change description

<SideBySideContainer>

<SideBySideColumn>

**In Strapi v4**

> The following list of configurations were only available from the environment:

Author: ```suggestion
The following list of configurations are only available from the environment:
```

</SideBySideColumn>

</SideBySideContainer>

<br />

## Migration

<MigrationIntro />

### Notes

The following environment variable names in Strapi v4 should now be defined in the [`/config/server.js` configuration file](/dev-docs/configurations/server):

| Environment variable name in Strapi v4 | Server configuration option in Strapi 5 |
|----------------------------------------|-----------------------------------------|
| `STRAPI_DISABLE_REMOTE_DATA_TRANSFER`  | `transfer.remote.enabled`               |
| `STRAPI_HIDE_STARTUP_MESSAGE`          | `logger.startup.enabled`                |
| `STRAPI_DISABLE_UPDATE_NOTIFICATION`   | `logger.updates.enabled`                |

### Migration steps

If you previously disabled one of the listed environment variables in your environment, update the `/config/server.js` by adding the appropriate values:

<Tabs groupId="js-ts">
<TabItem value="js" label="JavaScript">

```js title="/config/server.js"
module.exports = ({ env }) => ({
  // … other configuration options
  transfer: {
    remote: {

>       enabled: false, // disable remote data transfers instead of STRAPI_DISABLE_REMOTE_DATA_TRANSFER

Reviewer: ```suggestion
      enabled: !env.bool(STRAPI_DISABLE_REMOTE_DATA_TRANSFER), // if you want to continue using the old env variable
```
You may want to include an example to make it work exactly like v4, although I don't know if we want to encourage it since the point is that we're trying to reserve the STRAPI_ namespace for our own use. (so not necessarily accepting this suggestion, but adding something like this suggestion elsewhere)

Author: Thanks for the suggestion, Ben. I'd say that if we don't want to encourage it, it would be best to only document the best practices in our docs :)
